### DISCUSSION TITLE
blog plugin: customize blog posts URL

### DISCUSSION THREAD

**Discussion by vmihailenco** [vuepress-theme-hope/vuepress-theme-hope#2739] [Q&A] [answered]
For example, I have markdown files in `posts` folder so the generated URL is `/posts/filename.html`. Is it possible to remove `/posts` prefix from the generated URL, but keep markdown files in `posts` folder? So the URL `/posts/filename.html`. becomes just `/filename.html`.

I am using `blogPlugin({})` and can add whatever options are needed.

**Comment by Mister-Hope** [comment, marked as answer]
https://v2.vuepress.vuejs.org/reference/frontmatter.html#layout

**Reply by vmihailenco** [reply]
https://v2.vuepress.vuejs.org/reference/frontmatter.html#layout?

**Reply by Mister-Hope** [reply]
The heading under it: https://v2.vuepress.vuejs.org/reference/frontmatter.html#permalink

**Reply by vmihailenco** [reply]
That could work, but requires manually updating frontmatter for every page. I am lookin for somethings like:

```
      // only files under posts are articles
      filter: ({ filePathRelative }) => {
        return filePathRelative ? filePathRelative.startsWith('posts/') : false
      },
      page: (page) => {
        const PREFIX = "posts/"
        if (page.path.startsWith(PREFIX)) {
          // PREFIX is exactly at the beginning
          page.path = page.path.slice(PREFIX.length);
        }
      }
```

Is something like that possible?

**Reply by Mister-Hope** [reply]
You can just set page.permalink
